File: src/main/java/com/nicholasnassar/sfsapi/data/grades/FullGrade.java
package com.nicholasnassar.sfsapi.data.grades;

import io.vertx.core.json.JsonObject;

import java.util.List;

public class FullGrade {
    private final List<GradeAssignment> assignments;

    public FullGrade(List<GradeAssignment> assignments) {
        this.assignments = assignments;
    }

    public List<GradeAssignment> getAssignments() {
        return assignments;
    }

    public JsonObject asJson() {
        return new JsonObject().put("assignments", GradeAssignment.asJsonArray(assignments));
    }
}

File: src/main/java/com/nicholasnassar/sfsapi/data/grades/FullGrade.kt
package com.nicholasnassar.sfsapi.data.grades

import io.vertx.core.json.JsonObject

class FullGrade(val assignments: List<GradeAssignment>) {
    fun asJson(): JsonObject {
        return JsonObject().put("assignments", GradeAssignment.asJsonArray(assignments))
    }
}
